### PR TITLE
Move to cuDNN8; support dual CUDA versions (10.2/11.0)

### PR DESCRIPTION
**Pull request by jayfurmanek**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [ ] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

This bumps the cuDNN recipe up to cuDNN8. It also supports multiple CUDA versions: 10.2 and 11.0

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ mkdir -p $PREFIX/cuda/lib
 
 cp $RECIPE_DIR/LICENSE $PREFIX/doc/nvidia/cuDNN_LICENSE
 cp -r $SRC_DIR/cudnn/lib64/ $PREFIX/lib/
-cp -r $SRC_DIR/cudnn-dev/include/cudnn*.h $PREFIX/include/cudnn.h
+cp -r $SRC_DIR/cudnn-dev/include $PREFIX
```
Comment (bnemanich) [inline recipe/build.sh:21]: Is this going to be a problem? It looks like in the past we were converting a file name from cudnnSOMETHING.h to cudnn.h. Is CuDNN 8 setup differently?

Comment (jayfurmanek) [reply recipe/build.sh:21]: cudnn-dev7
```
./usr/include/cudnn_v7.h
./usr/lib64/libcudnn_static_v7.a
```
```
./usr/include/cudnn_adv_infer_v8.h
./usr/include/cudnn_adv_train_v8.h
./usr/include/cudnn_backend_v8.h
./usr/include/cudnn_cnn_infer_v8.h
./usr/include/cudnn_cnn_train_v8.h
./usr/include/cudnn_ops_infer_v8.h
./usr/include/cudnn_ops_train_v8.h
./usr/include/cudnn_v8.h
./usr/include/cudnn_version_v8.h
./usr/lib64/libcudnn_static_v8.a
```

So yea, perhaps a problem if we want to continue dropping that `_v` part.

Comment (jayfurmanek) [reply recipe/build.sh:21]: and I guess we'd want to drop the `_v8` on all of the .h files... I don't remember the reason why we currently do that for _v7.

**Review by npanpaliya — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,36 +1,63 @@
...
-{% set cudnn_version_x86 = "7.6.5.33-1" %}
 {% set cuda_version = cudatoolkit | replace('.*',"") %}
-{% set dev_sha256_ppc = "fdf3d23bfa3ba22bef9e508473ab20affc25d5b84614475d5cbf0b40573ab04b" %}
-{% set sha256_ppc = "378894e78783f6866e714d85afc199c69754da71cfe8312e0163b5397c713cf4" %}
-{% set dev_sha256_x86 = "644d9e4206371764893408bf961afceaa9ee63175c2670fe9f8577bf5397cf7a" %}
-{% set sha256_x86 = "317483416b6516c0fb3d38f103bdd315352625c2e16f75aa37bb2a9323dd34c1" %}
-
+{% set cudnnvars = {
+  "10.2": {
+    "os": "rhel7",
```
Comment (jayfurmanek) [inline recipe/meta.yaml:4]: Note that cuda10.2 is only supported on rhel7 for ppc64le, so we pull the `rhel7` version here for both platforms and `rhel8` below for the reverse reasons (Cuda11 is only supported on rhel8 for ppc64le).

There is more flexibility on x86-64, but we might as well keep these in sync.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,36 +1,63 @@
...
+    "checksums": {
+      "linux64": "24486c4069ad2b5b1fc8082db6bdbf6ad65825cac11a673a781a2289bad0aaf4",
+      "ppc64le": "c7456bc1d4ce982969a6b2aeae011275bcf8fbadf7dc7cbd136f5820f0ab5fe4",
+    },
+    "checksums-dev": {
+      "linux64": "5da88e8095c5ec7c911195ad930da8fc7dce213543e7360bdc073078190dfb6d",
```
Comment (npanpaliya) [inline recipe/meta.yaml:10]: So, linux-64 sha256sum values are corresponding to cudnn8 on x86 for cuda 10.2??

File: config/build-config.yaml [hunk 4]
```diff
@@ -1,6 +1,10 @@
 recipes:
+{% if cudatoolkit == '10.2' %}
```
Comment (npanpaliya) [inline config/build-config.yaml:2]: cudnn 7.6.5 should be built only on ppc for cudatoolkit 10.2.

Comment (jayfurmanek) [reply config/build-config.yaml:2]: It's probably best to keep the platforms in sync.

File: recipe/meta.yaml (hunk 3 above)
Comment (jayfurmanek) [reply recipe/meta.yaml]: yes

**Pull request merged**